osquery/core/watcher.cpp

```diff
@@ -360,14 +360,16 @@ Status WatcherRunner::isWatcherHealthy(const PlatformProcess& watcher,
 }
 
 QueryData WatcherRunner::getProcessRow(pid_t pid) const {
-#ifdef WIN32
-  pid = (pid == ULONG_MAX) ? -1 : pid;
-#endif
-
   // On Windows, pid_t = DWORD, which is unsigned. However invalidity
   // of processes is denoted by a pid_t of -1. We check for this
-  // by comparing the max value of DWORD, or ULONG_MAX
-  return SQL::selectAllFrom("processes", "pid", EQUALS, INTEGER(pid));
+  // by comparing the max value of DWORD, or ULONG_MAX, and then casting
+  // our query back to an int value, as ULONG_MAX causes boost exceptions
+  // as it's out of the range of an int.
+  int p = pid;
+  if (isPlatform(PlatformType::TYPE_WINDOWS)) {
+    p = (pid == ULONG_MAX) ? -1 : pid;
```

Comment:
On Windows, as we don't have a `pid_t` type, we use the more logical `DWORD` and cast it to a `pid_t` [here](https://github.com/facebook/osquery/blob/600a5d017accc17802b4e8acd99acdcfb0de2c27/include/osquery/system.h#L39). Unfortunately, we kinda rely on a query searching the processes table for `pid = -1` to assure as something hasn't started yet. In cases like these, we use `-1`, which ends up getting cast to be `ULONG_MAX`, which in Windows is an invalid PID value. I'm definitely down for other options here. PID values on Windows are unsigned longs, which means casting things to `int` might cause problems down the line if we ever hit a pid value that's greater than `INT MAX` but less than `ULONG_MAX`, however I think the odds of that are slim, and I'm not sure of a much better alternative aside from working harder to make the `-1` value work on windows by moving away from `DWORD`.